### PR TITLE
ckd datasets update

### PR DESCRIPTION
**Pull request by nollety**
# Description

Resolves eradiate/eradiate-issues#102

# Changes

* updated ckd absorption data sets, with the following main features:
  * based on `HITRAN2020` (previously a mix between HITRAN2016 and HITRAN2020)
  * correct use of ozone photodissociation absorption data (previously incorrect, affecting the range `400-650nm` in particular)
  * new `CO2` dimension, giving the carbon dioxide volume mixing ratio at sea level (representative of the amount of carbon dioxide in the entire atmosphere)
  * absorption data correspond to the 10-molecule version of the AFGL (1986) atmospheres, namely `H2O`, `CO2`, `O3`, `N2O`, `CO`, `CH4`, `O2`, `NO`, `SO2`, `NO2` (previously 7 molecules only were included)
* you can use the AFGL (1986) `us_standard` and `midlatitude_summer` variants (previously only `us_standard`)
* you can rescale `H2O`, `CO2` and `O3` (previously only `H2O` and `O3`)

## Notes about the FTP server update

I uploaded the new ckd data sets (gzipped) on our ftp server here: https://eradiate.eu/data/store/unstable/ckd/absorption.
The total upload size is 1.2 GB.
In order to not break previously running eradiate versions, I did not overwrite the previous ckd data sets. Instead, I appended a version number to the new data set filenames.
`src/eradiate/radprops/_util_ckd.py.PATHS` was updated with the addition of these filenames.

## Notes about the data submodule update

I updated the AFGL (1986) thermophysical properties data sets to add an `identifier` attribute. This identifier attribute is aligned with how the corresponding ckd absorption data sets are named.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [ ] The feature branch is rebased on the current state of the `next` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

:warning: we must pull eradiate/eradiate-data#6 before pulling this

### CODE REVIEW

**Comment by nollety** [comment]
At the moment, the PR does not pass the following regression tests:

* `tests/02_eradiate/03_regression/atmospheres/test_rpv_afgl1986.py`
* `tests/02_eradiate/03_regression/atmospheres/test_rpv_afgl1986_continental.py`

**Comment by nollety** [comment]
Since these regression tests run at 550 nm and include an absorbing molecular atmosphere, they are affected by the "ozone bug" which this PR aims to resolve.

**Comment by leroyvn** [comment]
> Since these regression tests run at 550 nm and include an absorbing molecular atmosphere, they are affected by the "ozone bug" which this PR aims to resolve.

Hi @nollety, this is the whole point of such tests. We'll check the new results and establish them as the reference.

**Comment by nollety** [comment]
Below are plots where I compare the regression tests results (`old` denotes the reference results and `new` denotes the current results, with the new ckd data sets):

* `tests/02_eradiate/03_regression/atmospheres/test_rpv_afgl1986.py`:
test_rpv_afgl1986

* `tests/02_eradiate/03_regression/atmospheres/test_rpv_afgl1986_continental.py`:
test_rpv_afgl1986_continental

**Comment by nollety** [comment]
@leroyvn, if you agree, this will update the reference results used in regression tests:
https://github.com/eradiate/eradiate-data/pull/6

File: src/eradiate/thermoprops/util.py [hunk 1]
```diff
@@ -137,6 +137,26 @@ def compute_mass_density_at_surface(ds, species):
     return m * compute_number_density_at_surface(ds=ds, species=species)
 
 
+def compute_volume_mixing_ratio_at_surface(ds, species):
```
Comment (leroyvn) [inline src/eradiate/thermoprops/util.py:140]: Type hints

File: src/eradiate/radprops/_afgl1986.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
 from ..thermoprops.util import (
     compute_column_mass_density,
     compute_column_number_density,
+    compute_volume_mixing_ratio_at_surface,
```
Comment (leroyvn) [inline src/eradiate/radprops/_afgl1986.py:25]: Side comment: I think this naming pattern ("compute...") is heavy and could be made more concise. After all, the fact that a computation occurs is made obvious, and this actually breaks the expressiveness of code using your functions. For example, compare
```python
x = compute_volume_mixing_ratio_at_surface(...)
```
and
```python
x = volume_mixing_ratio_at_surface(...)
```
This is much nicer to read IMO.

**Review by leroyvn — COMMENTED** [review]
Hi @nollety, not too much to say on my end. I made a couple of comments.

The data directory will have to be cleaned up eventually. If you want to make current data permanent, I think it's worth considering a transfer to the `stable` data store.

**Comment by leroyvn** [comment]
Sorry, I forgot to tick the right box: I approve the PR for merge after discussing points I raised.

**Comment by leroyvn** [comment]
Thanks @nollety, let's merge that.

**Pull request merged**